core/src/main/java/org/infinispan/container/versioning/irac/DefaultIracTombstoneManager.java

```diff
@@ -277,36 +279,58 @@ private CompletionStage<Void> performCleanup() {
       if (stopped) {
          return CompletableFutures.completedNull();
       }
-      boolean trace = log.isTraceEnabled();
+      var trace = log.isTraceEnabled();
 
       if (trace) {
          log.trace("[IRAC] Starting tombstone cleanup round.");
       }
 
       scheduler.onTaskStarted(tombstoneMap.size());
 
-      CompletionStage<Void> stage = Flowable.fromIterable(tombstoneMap.values())
+      var tombstoneStage = Flowable.fromIterable(tombstoneMap.values())
             .groupBy(this::classifyTombstone)
             // We are using flatMap to allow for actions to be done in parallel, but the requests in each action
             // must be performed sequentially
-            .flatMap(group -> {
-               switch (group.getKey()) {
-                  case REMOVE_TOMBSTONE:
-                     return removeAllTombstones(group);
-                  case NOTIFY_PRIMARY_OWNER:
-                     return notifyPrimaryOwner(group);
-                  case CHECK_REMOTE_SITE:
-                     return checkRemoteSite(group);
-                  case KEEP_TOMBSTONE:
-                  default:
-                     return Flowable.empty();
-               }
+            .flatMap(group -> switch (group.getKey()) {
+               case REMOVE_TOMBSTONE -> removeAllTombstones(group);
+               case NOTIFY_PRIMARY_OWNER -> notifyPrimaryOwner(group);
+               case CHECK_REMOTE_SITE -> checkRemoteSite(group);
+               default -> Flowable.empty();
```

Comment:
Technically this should also be `group.ignoreElements()` to ensure it was subscribed to. Guessing this code was never hit before.